src/script/standard.cpp

```diff
@@ -242,8 +242,13 @@ bool ExtractDestination(const CScript& scriptPubKey, CTxDestination& addressRet)
         addressRet = hash;
         return true;
     }
-    case TxoutType::WITNESS_UNKNOWN:
     case TxoutType::WITNESS_V1_TAPROOT: {
+        WitnessV1Taproot tap;
+        std::copy(vSolutions[1].begin(), vSolutions[1].end(), tap.begin());
```

Comment:
The explanation is just that this was easier, because while `TxoutType` did have a separation between unknown and taproot, CTxDestination did not. Making them both produce {version, program} as `vSolutions` was easier because it allowed handling them identically in `ExtractDestination`.

I've addressed this by first changing the Solver output for taproot outputs to be just {program}, like P2WPKH/P2WSH, in a new commit "c1445ee0d56e56d824bc79a0c9f1db26105ade1f", and then used that in the later commits.